### ISSUE TITLE
How can I use this package without calling static methods but rather construct injection?

### ISSUE THREAD

**Issue by mattkomarnicki** [ramsey/uuid#218]
Hi, take a look at below example:

```
<?php

abstract class AbstractTransformerService implements TransformerServiceInterface
{
    protected $uuid;

    public function __construct(
        UuidFactory $uuid
    ) {
        $this->uuid = $uuid;

        dd($this->uuid->uuid4()->toString()); // this works
        
        // how to use $this->uuid->isValid('foo'); ?
    }
}
```

So I can easily generate an `uuid` object and then cast it to a string.

I would like to also check if an `uuid` is valid. I don't want to use `Uuid::isValid()` (I want to avoid static methods).

How can I do this?

**Comment by ramsey** [comment]
Unfortunately, this is not available in the 3.x series, but it is available in the unreleased 4.x series.

You'll be able to do the following:

``` php
$uuidFactory->getValidator()->validate($uuid);
```

I'll keep this issue open so I can keep you updated on the status of the 4.x release.

**Comment by komarnicki** [comment]
> I'll keep this issue open so I can keep you updated on the status of the 4.x release.

Thanks a lot for being so keen to help. Slightly unusual. 🍺

**Comment by ramsey** [comment]
I've just released version 4.0.0-alpha1. In this version, you may do:

``` php
$factory = new \Ramsey\Uuid\UuidFactory();

$factory->getValidator()->validate('c0681b80-9603-478c-98ad-23428ffddba1');
```

You may also replace the default validator with a more strict validator:

``` php
$factory = new \Ramsey\Uuid\UuidFactory();
$factory->setValidator(new \Ramsey\Uuid\Rfc4122\Validator());
```
```

**Comment by mattkomarnicki** [comment]
Thanks @ramsey! 🤙🏻
